rust/rbac-registration/src/registration/cardano/mod.rs

```diff
@@ -345,13 +344,8 @@ impl RegistrationChainInner {
             context,
         )?;
 
-        let (purpose, registration, payment_history) = match cip509.consume() {
-            Ok(v) => v,
-            Err(e) => {
-                let error = format!("Invalid Cip509: {e:?}");
-                error!(error);
-                bail!(error);
-            },
+        let Ok((purpose, registration, payment_history)) = cip509.consume() else {
+            bail!("Unable to start a chain: invalid Cip509");
```

Comment:
Can this be better described?  What does it mean?

Reply:
This part was reworked.